core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -240,7 +248,7 @@ private[spark] object ReliableCheckpointRDD extends Logging {
       val bufferSize = sc.conf.getInt("spark.buffer.size", 65536)
       val partitionerFilePath = new Path(checkpointDirPath, checkpointPartitionerFileName)
       val fs = partitionerFilePath.getFileSystem(sc.hadoopConfiguration)
-      if (fs.exists(partitionerFilePath)) {
+      try {
```

Comment:
Seems to me that it can be pulled into the existing try/catch...its the only place in the codepath which can raise an FNFE; all existing failure modes (permissions, not a file, etc) would be caught there and log at warning...this just converts the FNFE to a log at debug event. 

-going to do that but adding the stack trace printing in the debug logs. Noisy, but your support team will love you on the one day it finally happens.